src/libasr/codegen/asr_to_wasm.cpp

```diff
@@ -579,18 +717,31 @@ class ASRToWASMVisitor : public ASR::BaseVisitor<ASRToWASMVisitor> {
         wasm::emit_export_mem(m_export_section, m_al, "memory", 0 /* mem_idx */);
         no_of_exports++;
 
-        declare_global_vars();
+        declare_global_var(wasm::type::i32, "cur_mem_loc", 0, true);
+        declare_global_var(wasm::type::f32, "tmp_reg_f32", 0, true);
+        declare_global_var(wasm::type::f64, "tmp_reg_f64", 0, true);
 
         emit_string(" ");
         emit_string("\n");
         emit_string("-");
         emit_string(".");
+        emit_string("(");
+        emit_string(")");
+        emit_string(",");
         digits_mem_loc = avail_mem_loc;
         for (int i = 0; i < 10; i++) {
             emit_string(std::to_string(i));
         }
         emit_print_int();
         emit_print_float();
+        emit_complex_add_32();
+        emit_complex_add_64();
+        emit_complex_sub_32();
+        emit_complex_sub_64();
+        emit_complex_mul_32();
+        emit_complex_mul_64();
+        emit_complex_abs_32();
+        emit_complex_abs_64();
```

Comment:
Ideally, these should be emitted into `wasm` code only when the particular operations are being used in the source code. 

**Concern/Challenge:** If we do not define theses functions initally and add them at the very end (that is after defining all the other functions), then we do not have any function index to refer to these functions when they are called during a call (while defining the other functions).

Reply:
I think I would keep a running index of all functions and just appending these at the end as needed; then once we are done, we convert the index to the function index at the start of the WASM binary file.

Reply:
I attempted to support it in [WASM: Support emitting functions based on requirement](https://github.com/lcompilers/lpython/pull/1527/commits/24b943aa2cd9cf3ef90e55af98765259b3dcfb51), and I'd love to get your feedback on it. Specifically, I'm wondering if this is what you had in mind, and if there are any changes you would like me to make. I'm open to any suggestions you have and would love to discuss this further with you.